app/server.js

```diff
@@ -1,32 +1,58 @@
 require('dotenv').config();
+const axios = require('axios');
 const config = require('./config');
 const { App } = require('./app');
 const { AuthProvider } = require('./authProvider');
 const { routes } = require('./routes');
 const { logger } = require('./logger');
 
-Object.keys(config).map((configKey) => {
-  if (config[configKey]) {
-    logger.info(`${configKey} set to ${config[configKey]}`);
-  } else {
-    logger.error(`${configKey} not set`);
-  }
-});
-
-const authProvider = new AuthProvider();
-const app = new App(authProvider).createApp();
-
-app.use(config.baseUrl ? config.baseUrl : '/', routes(authProvider));
-if (config.baseUrl) {
-  app.use('/', (req, res) => {
-    res.redirect(config.baseUrl);
+(async () => {
+  Object.keys(config).map((configKey) => {
+    if (config[configKey]) {
+      logger.info(`${configKey} set to ${config[configKey]}`);
+    } else {
+      logger.error(`${configKey} not set`);
+    }
   });
-}
-
-// Run application on configured port
-if (config.env === 'development') {
-  logger.info(`Public browse - \x1b[35m${config.appBaseUri}${config.baseUrl}/\x1b[0m`);
-} else {
-  logger.info(`App listening on port ${config.port} - Public browse`);
-}
-app.listen(config.port);
+
+  let intervalId;
+  let app;
```

Comment:
can you move these inside  `startApp`?

Reply:
Nope the intervalId is used outside of startApp. It is set in line 57 and then referenced in line 25 to end the polling.
If I move app inside startApp, everytime it does another poll this will get reset and it'll never come out of the loop 😢

Reply:
app can move in though?

Reply:
oh no it cant

Reply:
sad times

Reply:
Yeah I'm not happy with the amount of `let`s and also the lack of testing here. I think it can be tested 🤔 Also while I was playing about there is a library that can be used to promisify a function 
https://nodejs.org/api/util.html#util_util_promisify_original
this might tidy up the code base somewhat if that can be used. But didn't have the time

Reply:
tech debt